qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -0,0 +1,172 @@
+                                self.test_dirs_mount_path])
+
+        self.scratch_dirname = 'scratch'
+        self.mount_a.run_shell(['mkdir', self.scratch_dirname])
+        # read var name as "scratch dir's mount path"
+        self.scratch_dirs_mount_path = self.mount_a.client_remote.mkdtemp(
+            suffix=self.scratch_dirname)
+        self.mount_a.run_shell(['sudo','ln','-s',join(self.mount_a.mountpoint,
+                                                      self.scratch_dirname),
+                                self.scratch_dirs_mount_path])
+
+    def install_deps(self):
+        from teuthology.misc import get_system_type
+
+        args = ['sudo', 'install', '-y']
+
+        distro = get_system_type(self.mount_a.client_remote,
+                                 distro=True).lower()
+        if distro in ['redhatenterpriseserver', 'fedora', 'centos']:
+            deps = """acl attr automake bc dbench dump e2fsprogs fio \
+            gawk gcc indent libtool lvm2 make psmisc quota sed \
+            xfsdump xfsprogs \
+            libacl-devel libattr-devel libaio-devel libuuid-devel \
+            xfsprogs-devel btrfs-progs-devel python sqlite""".split()
+            deps_for_old_distros = "xfsprogs-qa-devel"
+
+            args.insert(1, 'yum')
+            args.extend(deps)
+            args.append(deps_for_old_distros)
+        elif distro == 'ubuntu':
+            deps = """xfslibs-dev uuid-dev libtool-bin \
+            e2fsprogs automake gcc libuuid1 quota attr libattr1-dev make \
+            libacl1-dev libaio-dev xfsprogs libgdbm-dev gawk fio dbench \
+            uuid-runtime python sqlite3""".split()
+
+            args.insert(1, 'apt-get')
+            args.extend(deps)
+        else:
+            raise RuntimeError('expected a yum based or a apt based system')
+
+        proc = self.mount_a.client_remote.run(args=args, omit_sudo=False)
+
+    def create_reqd_users(self):
+        self.mount_a.client_remote.run(args=['sudo', 'useradd', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'groupadd', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'useradd',
+                                             '123456-fsgqa'], omit_sudo=False,
+                                       check_status=False)
+
+    def write_local_config(self):
+        from os.path import join
+        from textwrap import dedent
+        from teuthology.misc import sudo_write_file
+
+        def get_a_mon_socket_addr():
+            from json import loads
+
+            mon_status = loads(self.fs.mon_manager.raw_cluster_cmd(
+                'mon_status'))
+            mon_socket_addr = mon_status['monmap']['mons'][0]['public_addrs']\
+                ['addrvec'][1]['addr']
+            return mon_socket_addr
```

Comment:
Since c09b82a80a392ccd0da7677c7b424ce5cd3fa5d6 this command is now a tell/asok command.

Better is to use `ceph --format=json mon dump` to get the addresses. I recommend making this a helper method in `qa/tasks/ceph_manager.py`. cc @liewegas 

See also this failure: `/ceph/teuthology-archive/pdonnell-2019-11-04_12:55:15-fs-wip-pdonnell-testing-20191104.085350-distro-basic-smithi/4472363/teuthology.log`

After fixing, please rerun through teuthology and link the results in this PR.

Reply:
http://qa-proxy.ceph.com/teuthology/pdonnell-2019-11-04_12:55:15-fs-wip-pdonnell-testing-20191104.085350-distro-basic-smithi/4472363/teuthology.log